process/block/metablock.go

```diff
@@ -520,6 +520,14 @@ func (mp *metaProcessor) indexBlock(
 		return
 	}
 
+	headerHash, err := core.CalculateHash(mp.marshalizer, mp.hasher, metaBlock)
```

Comment:
You already has headerHash computed in CommitBlock method from where you call this method. You can send this parameter and avoid to recalculate it again here

Reply:
done